### DISCUSSION TITLE
ObjectMapper in tests using DefaultDGSQueryExecutor

### DISCUSSION THREAD

**Discussion by jaroslaw-blazkow** [Netflix/dgs-framework#189] [Q&A]
Hi, I've got deserialization problems in tests after adding custom scalars. I've registered JavaTimeModule (JSR-310) into objectMapper bean. The problem is - in running application it works, but in tests still failing. I've debugged DefaultDGSQueryExecutor and turned out that it creates new instance of objectMapper and there's no simple solution to register additional modules or to reuse objectMapper from spring context. 
I've ended with creating custom implementation for DGSQueryExecutor where objectMapper is autowired from app context. 
The point is: am I missing something? If not - I suggest some changes ;)

**Comment by paulbakker** [comment]
Do you have some example code you can share to help reproduce the issue? I thought we have JavaTimeModule included already, but apparently something is still missing.

**Reply by jaroslaw-blazkow** [reply]
test:
`dgsQueryExecutor.executeAndExtractJsonPathAsObject( query, DATA, typeRef );
`

app:
```
  @Bean
  public ObjectMapper objectMapper( Jackson2ObjectMapperBuilder builder )
  {
    ObjectMapper objectMapper = builder.build();
    objectMapper.registerModule( ( new JavaTimeModule() ) );
    return objectMapper;
  }
```
  
 Compare objectMappers in debug (app vs test), they're different.

**Comment by jessepinkman9900** [comment]
I have a similar problem, it doesn't work in runtime as well.
I have defined custom scalar called `scalar DateTime`, and have written `@DgsScalar` similar to what's shown in the dgs documentation.
Currently getting this error
```
java.lang.IllegalArgumentException: Cannot construct instance of `java.time.OffsetDateTime` (no Creators, like default constructor, exist): cannot deserialize from Object value (no delegate- or property-based Creator)
```
Tried registering `new JavaTimeModule()` as well, it did not work.

From what I understand, it seems to be that the mapping form graphql mutation input to the generated input pojo is failing

**Reply by paulbakker** [reply]
This will be fixed by this PR: https://github.com/Netflix/dgs-framework/pull/271
